bridge-proxy/tests/bridge_proxy_blackbox_test.rs

```diff
@@ -18,123 +19,97 @@ use multiversx_sc::{
-
-        state
+impl BridgeProxyTestState {
+    fn new() -> Self {
+        let mut world = world();
+
+        world.account(OWNER_ADDRESS).nonce(1);
+        Self { world }
     }
 
     fn bridge_proxy_deploy(&mut self) -> &mut Self {
-        self.world.set_state_step(
-            SetStateStep::new()
-                .put_account(&self.owner, Account::new().nonce(1))
-                .new_address(&self.owner, 1, &self.bridge_proxy_contract),
-        );
-
-        let ic = &self.world.interpreter_context();
-        self.world.sc_deploy(
-            ScDeployStep::new()
-                .from(self.owner.clone())
-                .code(self.world.code_expression(BRIDGE_PROXY_PATH_EXPR))
-                .call(self.bridge_proxy_contract.init(ManagedAddress::zero())),
-        );
+        self.world.account(OWNER_ADDRESS).nonce(1);
+
+        self.world
+            .tx()
+            .from(OWNER_ADDRESS)
+            .typed(bridge_proxy_contract_proxy::BridgeProxyContractProxy)
+            .init(OptionalValue::Some(ManagedAddress::default()))
+            .code(BRIDGE_PROXY_PATH_EXPR)
+            .new_address(BRIDGE_PROXY_ADDRESS)
+            .run();
 
         self
     }
 
     fn deploy_adder(&mut self) -> &mut Self {
-        self.world.set_state_step(SetStateStep::new().new_address(
-            &self.owner,
-            2,
-            &self.adder_contract,
-        ));
-
-        self.world.sc_deploy(
-            ScDeployStep::new()
-                .from(self.owner.clone())
-                .code(self.world.code_expression(ADDER_BOGUS_PATH_EXPR))
-                .call(self.adder_contract.init(BigUint::zero())),
-        );
-
+        self.world
+            .tx()
+            .from(OWNER_ADDRESS)
+            .typed(adder_proxy::AdderProxy)
+            .init(BigUint::zero())
+            .code(ADDER_PATH_EXPR)
+            .new_address(ADDER_ADDRESS)
+            .run();
         self
     }
 }
 
 // #[test]
 fn deploy_deposit_test() {
-    let mut test = BridgeProxyTestState::setup();
-    let bridge_token_id_expr = "str:BRIDGE-123456"; // when specifying the token transfer
+    let mut test = BridgeProxyTestState::new();
+    let bridge_token_id_expr: &str = "str:BRIDGE-123456"; // when specifying the token transfer
 
     test.bridge_proxy_deploy();
     test.deploy_adder();
 
-    // test.world.sc_call(
-    //     ScCallStep::new()
-    //         .from(&test.owner)
-    //         .to(&test.bridge_proxy_contract)
-    //         .call(test.bridge_proxy_contract.unpause())
-    // );
-
     let mut args = ManagedVec::new();
     args.push(ManagedBuffer::from(&[5u8]));
 
     let eth_tx = EthTransaction {
-        from: test.eth_user.clone(),
-        to: ManagedAddress::from_address(&test.adder_contract.to_address()),
+        from: EthAddress {
+            raw_addr: ManagedByteArray::default(),
+        },
+        to: ManagedAddress::from_address(&AdderContract::new("sc:adder").to_address()),
```

Comment:
this can be written `ManagedAddress::from(ADDER_ADDRESS.eval_to_array())`
```suggestion
        to: ManagedAddress::from_address(&AdderContract::new("sc:adder").to_address()),
```